src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -554,3 +558,59 @@ export const updateFormSettings = (
     return okAsync(updatedForm.getSettings())
   })
 }
+
+/**
+ * Deletes form logic.
+ * @param form The original form to delete logic in
+ * @param logicId the logicId to delete
+ * @returns ok(true) on success
+ * @returns err(database errors) if db error is thrown during logic delete
+ * @returns err(LogicNotFoundError) if logicId does not exist on form
+ */
+export const deleteFormLogic = (
+  form: IPopulatedForm,
+  logicId: string,
+): Result<never, LogicNotFoundError> | ResultAsync<true, DatabaseError> => {
+  // First check if specified logic exists
+  if (!form.form_logics.some((logic) => logic.id === logicId)) {
+    logger.error({
+      message: 'Error occurred when deleting form logic',
+      meta: {
+        action: 'deleteFormLogic',
+        formId: form._id,
+        logicId,
+      },
+    })
+    return err(new LogicNotFoundError('logicId does not exist on form'))
+  }
+
+  // Remove specified logic and then update form logic
+  const updated_form_logic = form.form_logics.filter(
+    (logic) => logic.id !== logicId,
+  )
+  const ModelToUse = getFormModelByResponseMode(form.responseMode)
+
+  return ResultAsync.fromPromise(
+    ModelToUse.findByIdAndUpdate(
+      form._id,
+      { form_logics: updated_form_logic },
```

Comment:
should we use mongo's [`$pull`](https://mongoosejs.com/docs/api/array.html#mongoosearray_MongooseArray-pull) operator instead? 

```ts
{ 
  form_logics: {
    $pull: { _id: logicId }
  }
}
```

Reply:
edited - but i think form_logics should be nested inside pull :)